ta_services/secure_key_services/src/handle.c

```diff
@@ -49,6 +50,10 @@ uint32_t handle_get(struct handle_db *db, void *ptr)
 		new_max_ptrs = db->max_ptrs * 2;
 	else
 		new_max_ptrs = HANDLE_DB_INITIAL_MAX_PTRS;
+
+	if (new_max_ptrs > (UINT32_MAX / 2))
+		return 0;
```

Comment:
Could we do without this restriction?

Reply:
for the C-code sequence i proposed, handles above UINT32_MAX/2 overflow the subtraction and lead to corrupted validation mask values. This is why i force IDs (handles) to be lower than this max value. And I guess handles will never reach such value.